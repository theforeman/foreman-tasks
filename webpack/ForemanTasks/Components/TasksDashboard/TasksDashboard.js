import React from 'react';
import PropTypes from 'prop-types';

const TasksDashboard = ({ children, className }) => (
  <div className={className}>{children}</div>
);

TasksDashboard.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

TasksDashboard.defaultProps = {
  className: '',
  children: null,
};

export default TasksDashboard;
