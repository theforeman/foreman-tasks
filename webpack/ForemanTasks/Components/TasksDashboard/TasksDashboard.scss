@import 'foremanReact/common/variables';
@import 'foremanReact/common/scss/mixins';

@mixin create-tasks-dashboard-column($columns: 12, $screen-min: 0, $gutter: $grid-gutter-width) {
  @media (min-width: $screen-min) {
    width: percentage(($columns / $grid-columns));
    float: left;
    position: relative;
    min-height: 1px;
    padding-right: ($gutter / 2);
    padding-left: ($gutter / 2);
  }
}

.tasks-dashboard-grid {
  min-height: 200px;
  background-color: #f5f5f5;
  margin: 5px -60px 20px -60px;
  padding: 20px 50px;

  .row > div {
    @include create-tasks-dashboard-column(12, 0);
    @include create-tasks-dashboard-column(6, 900px);
    @include create-tasks-dashboard-column(3, 1500px);
  }
}
