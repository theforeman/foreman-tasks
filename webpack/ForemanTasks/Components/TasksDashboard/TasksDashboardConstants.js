import { translate as __ } from 'foremanReact/common/I18n';

export const TASKS_DASHBOARD_AVAILABLE_SEARCH_QUERY_TYPES = {
  RUNNING: 'RUNNING',
  PAUSED: 'PAUSED',
  STOPPED: 'STOPPED',
  SCHEDULED: 'SCHEDULED',
  LAST: 'LAST',
  OVER: 'OVER',
  ERROR: 'ERROR',
  WARNING: 'WARNING',
  SUCCESS: 'SUCCESS',
};

export const TASKS_DASHBOARD_AVAILABLE_TIME_PERIODS = {
  H24: 'H24',
  H12: 'H12',
  WEEK: 'WEEK',
};

export const TASKS_DASHBOARD_AVAILABLE_TIME_PERIODS_TEXT = {
  [TASKS_DASHBOARD_AVAILABLE_TIME_PERIODS.H24]: __('24h'),
  [TASKS_DASHBOARD_AVAILABLE_TIME_PERIODS.H12]: __('12h'),
  [TASKS_DASHBOARD_AVAILABLE_TIME_PERIODS.WEEK]: __('week'),
};

export const FOREMAN_TASKS_DASHBOARD_UPDATE_SEARCH_QUERY =
  'FOREMAN_TASKS_DASHBOARD_UPDATE_SEARCH_QUERY';

export const FOREMAN_TASKS_DASHBOARD_UPDATE_TIME_PERIOD =
  'FOREMAN_TASKS_DASHBOARD_UPDATE_TIME_PERIOD';
