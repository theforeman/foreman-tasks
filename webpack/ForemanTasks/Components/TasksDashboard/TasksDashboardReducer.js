import Immutable from 'seamless-immutable';
import {
  TASKS_DASHBOARD_AVAILABLE_TIME_PERIODS,
  FOREMAN_TASKS_DASHBOARD_UPDATE_TIME_PERIOD,
} from './TasksDashboardConstants';

const initialState = Immutable({
  timePeriod: TASKS_DASHBOARD_AVAILABLE_TIME_PERIODS.H24,
});

export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {
    case FOREMAN_TASKS_DASHBOARD_UPDATE_TIME_PERIOD:
      return state.set('timePeriod', payload);
    default:
      return state;
  }
};
