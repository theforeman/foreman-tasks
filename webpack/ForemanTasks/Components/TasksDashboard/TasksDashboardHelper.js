import { TASKS_DASHBOARD_AVAILABLE_TIMES_TEXT } from './TasksDashboardConstants';

export const getTimeText = time => TASKS_DASHBOARD_AVAILABLE_TIMES_TEXT[time];
