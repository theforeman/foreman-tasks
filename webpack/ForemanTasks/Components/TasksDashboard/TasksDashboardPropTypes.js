import PropTypes from 'prop-types';
import { TASKS_DASHBOARD_AVAILABLE_TIMES } from './TasksDashboardConstants';

export const timePropType = PropTypes.oneOf(
  Object.values(TASKS_DASHBOARD_AVAILABLE_TIMES)
);

export const queryPropType = PropTypes.shape({
  state: PropTypes.string,
  result: PropTypes.string,
  mode: PropTypes.string,
  time: timePropType,
});
