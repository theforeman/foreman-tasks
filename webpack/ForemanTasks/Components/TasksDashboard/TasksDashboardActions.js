import { resolveSearchQuery } from 'foremanReact/components/SearchBar/SearchBarHelpers';

import { createSearchQueryByType } from './TasksDashboardHelper';
import {
  FOREMAN_TASKS_DASHBOARD_UPDATE_TIME_PERIOD,
  FOREMAN_TASKS_DASHBOARD_UPDATE_SEARCH_QUERY,
} from './TasksDashboardConstants';

export const updateTimePeriod = timePeriod => ({
  type: FOREMAN_TASKS_DASHBOARD_UPDATE_TIME_PERIOD,
  payload: timePeriod,
});

export const updateSearchQuery = (types, timePeriod) => dispatch => {
  const searchQuery = types
    .map(type => createSearchQueryByType(type, timePeriod))
    .join(' and ');

  dispatch({
    type: FOREMAN_TASKS_DASHBOARD_UPDATE_SEARCH_QUERY,
    payload: searchQuery,
  });

  resolveSearchQuery(searchQuery);
};
