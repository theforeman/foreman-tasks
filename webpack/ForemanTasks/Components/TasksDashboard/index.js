import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import TasksDashboard from './TasksDashboard';
import * as actions from './TasksDashboardActions';
import reducer from './TasksDashboardReducer';
import {
  selectTime,
  selectQuery,
  selectTasksSummary,
} from './TasksDashboardSelectors';

const mapStateToProps = state => ({
  time: selectTime(state),
  query: selectQuery(state),
  tasksSummary: selectTasksSummary(state),
});

const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

export const reducers = { tasksDashboard: reducer };

export default connect(mapStateToProps, mapDispatchToProps)(TasksDashboard);
