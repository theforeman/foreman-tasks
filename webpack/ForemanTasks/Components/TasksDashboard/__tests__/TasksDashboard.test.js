import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import TasksDashboard from '../TasksDashboard';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('TasksDashboard', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TasksDashboard, fixtures));
});
