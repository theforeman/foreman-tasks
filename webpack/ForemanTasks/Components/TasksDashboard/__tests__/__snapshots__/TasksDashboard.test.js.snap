// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksDashboard rendering render without Props 1`] = `
<div
  className=""
/>
`;
